packages/chat/src/ChatClient.ts

```diff
@@ -2139,15 +2140,21 @@ export class ChatClient extends IrcClient {
 		if (attributes.replyTo) {
 			tags['reply-parent-msg-id'] = extractMessageId(attributes.replyTo);
 		}
-		await this._messageRateLimiter.request(
-			{
-				type: 'say',
-				channel: toChannelName(channel),
-				text,
-				tags
-			},
-			rateLimiterOptions
+
+		// eslint-disable-next-line @typescript-eslint/promise-function-async
+		const textReqs = autoSplitterWithSpaces(text, 500).map(msg =>
```

Comment:
The place for an intermediate variable is weird here - instead, I'd save the splitter result in a variable and then call `Promise.all` on the `.map` result directly.